pkg/settings/map.go

```diff
@@ -45,16 +45,20 @@ func (s *SettingMap[T]) GetOrDefault(ctx context.Context, g Getter) (value T, er
 	if err != nil {
 		return s.Default.DefaultValue, fmt.Errorf("failed to get value from context: %w", err)
 	}
-	if g == nil {
+	//TODO confirm via test
+	valueOrDefault := func() (T, error) {
```

Comment:
This logic was only applied when the SettingsGetter was nil, but even if DefaultGetter is nil (like our tests), core wraps it in an `AtomicSettings` getter, so it was bypassed. Instead we need to apply it in every default case.